pkg/types/encrypted_private_key.go

```diff
@@ -29,74 +31,81 @@ func (a *EncryptedPrivateKey) validate() error {
-	return a.MarshalBinary()
+func (e EncryptedPrivateKey) MarshalYAML() (interface{}, error) {
+	return e.MarshalBinary()
 }
 
 // Custom YAML unmarshaller for EncryptedPrivateKey
-func (a *EncryptedPrivateKey) UnmarshalYAML(unmarshal func(interface{}) error) error {
+func (e *EncryptedPrivateKey) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	var data []byte
 	if err := unmarshal(&data); err != nil {
 		return err
 	}
 
-	return a.UnmarshalBinary(data)
+	return e.UnmarshalBinary(data)
 }
 
 // MarshalText overloads the TextMarshaler interface for EncryptedPrivateKey.
-func (a *EncryptedPrivateKey) MarshalText() ([]byte, error) {
-	if err := a.validate(); err != nil {
+func (e *EncryptedPrivateKey) MarshalText() ([]byte, error) {
+	if err := e.validate(); err != nil {
 		return nil, err
 	}
-	return a.Object.MarshalText()
+	return e.Object.MarshalText()
 }
 
 // UnmarshalText overloads the TextUnmarshaler interface for EncryptedPrivateKey.
-func (a *EncryptedPrivateKey) UnmarshalText(text []byte) error {
-	if err := a.Object.UnmarshalText(text); err != nil {
+func (e *EncryptedPrivateKey) UnmarshalText(text []byte) error {
+	if err := e.Object.UnmarshalText(text); err != nil {
 		return err
 	}
 
-	return a.validate()
+	return e.validate()
 }
 
 // MarshalBinary overloads the BinaryMarshaler interface for EncryptedPrivateKey.
-func (a *EncryptedPrivateKey) MarshalBinary() ([]byte, error) {
-	if err := a.validate(); err != nil {
+func (e *EncryptedPrivateKey) MarshalBinary() ([]byte, error) {
+	if err := e.validate(); err != nil {
 		return nil, err
 	}
 
-	return a.Object.MarshalBinary()
+	return e.Object.MarshalBinary()
 }
 
 // UnmarshalBinary overloads the BinaryUnmarshaler interface for EncryptedPrivateKey.
-func (a *EncryptedPrivateKey) UnmarshalBinary(data []byte) error {
-	a.Object.Version = 0x00 // We can't know the version at this stage. The version is in the plain text. `data` is ciphered.
-	a.Object.Data = data
+func (e *EncryptedPrivateKey) UnmarshalBinary(data []byte) error {
+	if e.Object == nil {
+		e.Object = &object.Object{
+			Kind:    object.EncryptedPrivateKey,
+			Version: 0x00,
+			Data:    nil,
+		}
+	}
+
+	e.Object.Version = 0x00 // We can't know the version at this stage. The version is in the plain text. `data` is ciphered.
+	e.Object.Data = data
 
-	return a.validate()
+	return e.validate()
 }
 
-// Sign signs the given data using the private key.
-// password is destroyed just after being used even if an error occurs.
+// Sign signs the given data using the private key. Password is destroyed.
 func (e *EncryptedPrivateKey) Sign(password *memguard.LockedBuffer, salt, nonce, data []byte) ([]byte, error) {
 	digest := blake3.Sum256(data)
 
-	privateKeyInClear, err := PrivateKey(password, salt, nonce, e.Data)
+	privateKeyInClear, err := privateKey(password, salt, nonce, e.Data)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("Sign: %w", err)
```

Comment:
Error message should start with capital letters in go. Furthermore your error is not giving context, it simply contains the name of the function.

my suggestion:
```suggestion
		return nil, fmt.Errorf("failed to get private key: %w", err)
```